### PR TITLE
 hide the whole AppMenu for non members.

### LINKED ISSUE #38: [UserCalendar] Hide AppMenu for Non Admins and Members on Mobile and Tablets
Hide the AppMenu containing TeamMembers Component on Calendar Page if the user is non admin or non member.

### PR DESCRIPTION
**Pull request by samagragupta**
## Issue
fixes: #38 

## Solution
Given access to admin and members

## Checklist
*PR will only be reviewed if the checklist is completed*
- [x] I have done extensive testing for my changes.
- [x] I have self-reviewed my PR and removed all un-necessary changes.
- [x] My changes works well on both desktop and mobile environments.
- [x] I have made sure that my changes solves the actual problem mentioned in the issue.
- [x] The PR contains only 1 commit and have been updated with `master`.

### CODE REVIEW

**Comment by 0xnmn** [comment]
The PR title and description makes no sense.

**Comment by samagragupta** [comment]
@hereisnaman Done

File: app/containers/UserCalendar/index.js [hunk 1]
```diff
@@ -8,9 +8,11 @@ import createPage from 'Utils/createPage';
 import AppContent from 'Components/AppContent';
 import Content from './Content';
 import TeamMembers from './TeamMembers';
+import useViewer from 'Hooks/useViewer';
 import authHelper from 'Utils/authHelper';
 
 const UserCalendar = ({ loading, viewer, user, router, refetch }) => {
+  const viewers = useViewer();
```
Comment (0xnmn) [inline app/containers/UserCalendar/index.js]: viewer is already present in the scope, not need to do this.

File: app/containers/UserCalendar/index.js [hunk 2]
```diff
@@ -34,21 +36,25 @@ const UserCalendar = ({ loading, viewer, user, router, refetch }) => {
...
-                id === viewer.user.id
-                  ? Router.push(...paths.me.calendar())
-                  : Router.push(...paths.users.calendar({ id }))
-              }
-              selected={selectedUser}
-            />
-          )}
-        </AppMenu.Context.Consumer>
-      </AppMenu>
+      {['Admin', 'Member'].includes(viewers.user.role) && (
```
Comment (0xnmn) [inline app/containers/UserCalendar/index.js]: Use [AuthHelper](https://github.com/coding-blocks/gondor/blob/master/app/utils/authHelper.js) here.

Also if you remove AppMenu, you also need to not wrap the Content with AppContent.

**Review by 0xnmn — APPROVED** [review]

**Review by 0xnmn — CHANGES_REQUESTED** [review]
fix

**Comment by samagragupta** [comment]
@hereisnaman Done

File: app/containers/UserCalendar/index.js [hunk 3]
```diff
@@ -30,14 +30,14 @@ const UserCalendar = ({ loading, viewer, user, router, refetch }) => {
...
-      </AppContent>
-      {['Admin', 'Member'].includes(viewers.user.role) && (
+      {authHelper.isMember(viewers) && (
         <>
+          <Head>
+            <title>Calendar | CodingBlocks</title>
+          </Head>
+          <AppContent>
+            <Content user={selectedUser} />
+          </AppContent>
```
Comment (0xnmn) [inline app/containers/UserCalendar/index.js:46]: You just need to stop wrapping the Content with AppContent if the viewer is not a member.

**Review by 0xnmn — CHANGES_REQUESTED** [review]

File: app/containers/UserCalendar/index.js (hunk 3 above)
Comment (samagragupta) [reply app/containers/UserCalendar/index.js:46]: Thats I have done at line 56.

Comment (0xnmn) [reply app/containers/UserCalendar/index.js:46]: We should not duplicate the whole bunch of code. 

Just the Content part. Also if the user is not a member, wrap the Content with <div class="row">

**Review by 0xnmn — CHANGES_REQUESTED** [review]

**Comment by samagragupta** [comment]
@hereisnaman Is it fine?

File: app/containers/UserCalendar/index.js [hunk 4]
```diff
@@ -31,24 +31,33 @@ const UserCalendar = ({ loading, viewer, user, router, refetch }) => {
...
-            />
-          )}
-        </AppMenu.Context.Consumer>
-      </AppMenu>
+      {!authHelper.isMember(viewer) && (
+        <>
+          <Content user={selectedUser} />
+        </>
+      )}
+      {authHelper.isMember(viewer) && (
```
Comment (0xnmn) [inline app/containers/UserCalendar/index.js]: {!authHelper.isMember(viewer) ? (
        <div className="row">
          <Content user={selectedUser} />
        </div>
      ) : (....)

**Review by 0xnmn — CHANGES_REQUESTED** [review]
require small fix

**Comment by 0xnmn** [comment]
This is a fair warning, multiple times you have not read the reviews carefully and miss something or other.

From next time I'll close the PR if found this behavior repeated. 


```javascript
!authHelper.isMember(viewer) ? (
    <div className="row">
      <Content user={selectedUser} />
    </div>
  ) : (....)
```

You need to wrap the Content with div for not logged in users. I don't think you have tested it properly otherwise you would have found the issue.

**Comment by samagragupta** [comment]
@hereisnaman wrapping it inside the div would be like this:

Capture
 And without wrapping it, it would be like this:


If you still think that one is better. I will make the change.

**Comment by 0xnmn** [comment]
> @hereisnaman wrapping it inside the div would be like this:
> 
> Capture
> And without wrapping it, it would be like this:
> 
> 
> If you still think that one is better. I will make the change.

You'll also need to wrap it with col-12 inside row. That I left for you to figure out.

**Comment by samagragupta** [comment]
@hereisnaman Done

**Review by 0xnmn — APPROVED** [review]

**Pull request merged**